### PR TITLE
Custom loggers

### PR DESCRIPTION
**Pull request by bswck**
Re-PR for a complete code review

### CODE REVIEW

**Review by bswck — COMMENTED** [review]
Be sure to reformat all the files with `black` and optionally resolve things that `flake8` complains about (or `pylint` if you are meticulous).

File: wdp/custom_loggers/config.py [hunk 1]
```diff
@@ -1,33 +0,0 @@
-# ----------- LOGS FOLDER -----------
-
-SUB_FOLDER = "log"
-
-# ----------- GENERAL LOG -----------
-
-GENERAL_LOG_FORMAT = "%(asctime)s - %(name)s - %(levelname)s - %(message)s"
-GENERAL_LOG_FORMAT_S = "%(asctime)s - %(name)s - %(levelname)s - Message: %(message)s - from module %(module)s - " \
```
[the remark this answers is not in the document]
Comment (tomekkurzydlak) [reply wdp/custom_loggers/config.py:8]: Haha, that was a temporary solution, just forgot to change it - S stands for second (string) ;p

File: wdp/custom_loggers/profiler.py [hunk 2]
```diff
@@ -1,174 +0,0 @@
...
-    print(f"Memory usage: {memory:.2f} MB\n")
-    print("=" * 10, f"Generated {now_str}", "=" * 10)
-
-
-def log_to_json(func: callable, memory: float, stats: pstats.Stats) -> None:
-    """ Logs profiler statistics and memory usage to a JSON file """
-    string = {
-        "captured_data": {
-            "timestamp": timestamp,
-            "log_time": now_str,
```
[the remark this answers is not in the document]
Comment (tomekkurzydlak) [reply wdp/custom_loggers/profiler.py:74]: Well I don't think it matters here as it's just a dump to json, and what we need here is a function execution time what is calculated right below "log_time", but yeah, will call this function right before a dump/print

File: wdp/custom_loggers/general.py [hunk 3]
```diff
@@ -1,174 +0,0 @@
...
-        else:
-            logger_format = log_format_s
-        formatter = logging.Formatter(logger_format)
-        for handler in logger.handlers:
-            handler.setFormatter(formatter)
-        logger.setLevel(level_to_log)
-        return logger
-
-
-def get_default_logger(use_decorator_format: bool = True) -> logging.Logger:
```
[the remark this answers is not in the document]
Comment (tomekkurzydlak) [reply wdp/custom_loggers/general.py:90]: It's all about dealing with two string formats. When it's a default logger, called with a decorator, I want to format its string with GENERAL_LOG_FORMAT, as I get all the infos directly from the passed function. When the logger object is created without decorator, directly using the generic class, I want to use logging's library placeholders such as {lineno} or {funcName}

File: wdp/custom_loggers/general.py [hunk 4]
```diff
@@ -1,174 +0,0 @@
...
-    return GenericLogger().get_logger(use_decorator_format=use_decorator_format)
-
-
-def log(
-    func_to_decorate=None, *,
-    my_logger: Union[GenericLogger, logging.Logger] = None,
```
[the remark this answers is not in the document]
Comment (tomekkurzydlak) [reply wdp/custom_loggers/general.py:96]: I believe I use this name already, but can go with passed_logger or new_logger

File: wdp/custom_loggers/general.py [hunk 5]
```diff
@@ -1,174 +0,0 @@
...
-            frame = inspect.currentframe().f_back
-            filename = inspect.getframeinfo(frame).filename
-            module_name = filename.split('/')[-1].split('.')[0]
-            line_number = inspect.getframeinfo(frame).lineno
-            lineno = inspect.getsourcelines(func)[1]
-
-            logger: logging.Logger = get_default_logger(use_decorator_format=False)
-            try:
-                if not my_logger:
-                    first_args = next(iter(args), None)
```
[the remark this answers is not in the document]
Comment (bswck) [reply wdp/custom_loggers/general.py:126]: How on earth would you get an unsliceable `args` if it's a positional variadic parameter?
Definitely `next()` is an overkill here. And even if you ever need to take 1 or more elements from an iterator, `itertools.islice()` or `more_itertools.take()` are the best options to choose from.

**Pull request closed without merging**